frontend/app/components/card.tsx

```diff
@@ -0,0 +1,161 @@
+import type { ComponentProps, HTMLElementType, KeyboardEvent, MouseEvent } from 'react';
+
+import type { To } from 'react-router';
+
+import type { FlipProp, IconProp } from '@fortawesome/fontawesome-svg-core';
+import { faChevronRight } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import { AppLink } from '~/components/links';
+import type { I18nRouteFile } from '~/i18n-routes';
+import { cn } from '~/utils/tailwind-utils';
+
+type OptionalAppLink = Omit<ComponentProps<typeof AppLink>, 'to' | 'lang'> & {
+  to?: To;
+  file?: I18nRouteFile;
+};
+
+type CardProps = OptionalAppLink & {
+  /**
+   * Optional FontAwesome icon to be displayed.
+   */
+  icon?: IconProp;
+
+  /**
+   * The flip property for the icon.
+   */
+  iconFlip?: FlipProp;
+
+  /**
+   * Optional image URL to be displayed at the top of the card.
+   */
+  image?: string;
+
+  /**
+   * Alternative text for the image.
+   */
+  imageAlt?: string;
+
+  /**
+   * Optional tag text to be displayed at the top-left corner of the card.
+   */
+  tag?: string;
+
+  /**
+   * The title text to be displayed.
+   */
+  title: string;
+
+  /**
+   *  The HTML element to be used for the title
+   *  @default 'h3'
+   */
+  titleAs?: ExtractStrict<HTMLElementType, 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p'>;
+};
+
+/**
+ * Base component for the Card component
+ * Returns an AppLink when the 'To' or 'File' prop is provided.
+ * Otherwise, returns an Anchor tag that functions as a button.
+ */
+function CardBase(props: OptionalAppLink) {
+  const { to, file, disabled } = props;
+  const baseClassName = `select-none rounded-sm ${disabled ? 'bg-white' : 'bg-blue-900'}`;
+  if (to || file) {
+    return (
+      <div className={baseClassName}>
+        <AppLink {...(props as ComponentProps<typeof AppLink>)} />
+      </div>
+    );
+  } else {
```

Comment:
Nitpick, the `else {}` is unnecessary here.

``` typescript
const baseClassName = `select-none rounded-sm ${disabled ? 'bg-white' : 'bg-blue-900'}`;

if (to || file) {
  return // component
}

// rest of code
```

Reply:
I was thinking a bit more about this and maybe for the sake of just getting it done, another separate component would be fine.